### ISSUE TITLE
update combobox component

### ISSUE THREAD

**Issue by MattL75** [SAP/fundamental-ngx#742]
Currently, our dropdown is made from a fd-popover component. This can be confusing, and also prevents us from **natively supporting Angular forms**, leaving the responsibility up to the user.

Separating the dropdown to its own component would also allow us to **remove the `isDropdown` input** from the Popover component (and associated style inputs), something which I would personally love to do.

@droshev 

As for the combobox, we have a new plan.

* Deprecate the search input, include warnings about this for devMode in the code.
* Make combobox a tad more flexible (templates for items, ng-content in the popover) and have that be the new 'search input'.
* Make the shellbar search input a standalone thing when we remove search input.

So basically we have these subtasks.

- [ ] Deprecate search input in favor of combobox.
- [ ] Make combobox standalone.
- [ ] Improve combobox functionality a little bit. Add template possibility for items.
- [ ] Make a separate 'search input' component for use inside shellbar.
- [ ] Make a select page where you can see the fd-select and native selects.

And eventually...

- [ ] Remove search input completely.

**Comment by droshev** [comment]
Some requirements:
- Inputs:
- - _container_ - a selector specifying the element the dropdown should be appended to. The default value is the body
- - _auto-close_ - default dropdown auto-closing behaviour. Ex. select an item (insideClick), or press the ESC key
- - _disabled_ - an option to disable the dropdown
- Outputs:
- - _on-toggle(open)_ - an event fired when the dropdown menu is opened or closed.
- Methods:
- - _open_ - manual triggering (click) and programmatically
- - _close_ - manual triggering (click) and programmatically
- - _toggle_ - manual triggering (click) and programmatically
- - _isOpen_ - check if the menu is open or not

**Comment by MattL75** [comment]
@droshev Some comments.

Input for specifying the element will be called *appentTo*. Container is not accurate as elements are appended to the selected element, not below the element.

Auto-close would not be flexible enough of an input. I'm instead thinking of separating this behaviour into things like closeOnOutsideClick, clickOnEscKey or closeOnSelect.

The body being removed from the DOM is automatic through our Popover directive, or our Popover component, so there's no need to worry about this.

Why would we want to provide a feature for rendering a templateRef? If users have access to their own content through ng-content then this is very pointless. Were you planning on making Dropdown a directive? If so, we could also include loading from a component.
